src/org/jetbrains/plugins/scala/codeInspection/functionExpressions/UnnecessaryPartialFunctionInspection.scala

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.plugins.scala.codeInspection.functionExpressions
+
+import com.intellij.codeInspection.{ProblemHighlightType, ProblemsHolder}
+import com.intellij.psi.{PsiClass, PsiElement}
+import org.jetbrains.plugins.scala.codeInspection.functionExpressions.UnnecessaryPartialFunctionInspection._
+import org.jetbrains.plugins.scala.codeInspection.{AbstractInspection, InspectionBundle}
+import org.jetbrains.plugins.scala.lang.psi.api.base.patterns._
+import org.jetbrains.plugins.scala.lang.psi.api.expr.ScBlockExpr
+import org.jetbrains.plugins.scala.lang.psi.impl.ScalaPsiManager
+import org.jetbrains.plugins.scala.lang.psi.types.{ScTypeParameterType, _}
+
+object UnnecessaryPartialFunctionInspection {
+  private val inspectionId = "UnnecessaryPartialFunction"
+  private val PartialFunctionClassName = classOf[PartialFunction[_, _]].getCanonicalName
+  private val Function1ClassName = classOf[(_) => _].getCanonicalName
+  val inspectionName = InspectionBundle.message("unnecessary.partial.function")
+}
+
+class UnnecessaryPartialFunctionInspection
+  extends AbstractInspection(inspectionId, inspectionName){
+
+  override def actionFor(holder: ProblemsHolder): PartialFunction[PsiElement, Any] = {
+    case expression: ScBlockExpr =>
+      def isNotPartialFunction(expectedType: ScType) =
+        findPartialFunctionType(holder).exists(!expectedType.conforms(_))
+      def conformsTo(expectedType: ScType) = (inputType: ScType, resultType: ScType) =>
+        findType(holder, Function1ClassName, _ => Seq(inputType, resultType)).exists(_.conforms(expectedType))
+
+      for{
+        expectedExpressionType <- expression.expectedType()
+        if isNotPartialFunction(expectedExpressionType)
+        Seq(singleCaseClause) <- expression.caseClauses.map(_.caseClauses)
+        if canBeConvertedToFunction(singleCaseClause, conformsTo(expectedExpressionType))
+        caseKeyword <- singleCaseClause.firstChild
+      } holder.registerProblem(
+          caseKeyword,
+          inspectionName,
+          ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+          new UnnecessaryPartialFunctionQuickFix(singleCaseClause, expression))
+  }
+
```

Comment:
`ProblemsHolder` is intended for passing problems around. It looks strange as an argument of methods like `findType`.